server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsAction.java

```diff
@@ -103,19 +106,32 @@ protected ClusterBlockException checkBlock(Request request, ClusterState state)
 
     public static class Request extends MasterNodeReadRequest<Request> {
 
-        public Request(TimeValue masterNodeTimeout, TaskId parentTaskId) {
+        private final Set<String> metrics;
+
+        public Request(TimeValue masterNodeTimeout, TaskId parentTaskId, Set<String> metrics) {
```

Comment:
I think passing entire metrics set here makes it miss leading. May be change class name to TransportGetMasterStats so it can handle a range of different metrics. Or pass few specific boolean flags Fs and Allocations to TransportGetAllocationStats. YAGNI style would be second option

Reply:
I would like to not use boolean flags as adding a new metric would require introducing a new flag and a new transport metric. I also chose to send the entire set as it was easier to do and to make bwc easier if/when in future more metrics (new or existing ones) are red from master (please remember that FS required the same mechanism shortly after allocation metric introduced it).

Reply:
Yeah, I understand convenience. As a reader I found it's confusing that TransportGetAllocationStatsAction can have a set of irrelevant metrics like SCRIPT_CACHE and JVM, that will be silently dropped. But it's in same spirit as surprising "FS".